### ISSUE TITLE
consider non-UTF8 characters as a linting issue to report instead of skipping the file

### ISSUE THREAD

**Issue by guillp** [rvben/rumdl#904]
### Problem or motivation

I am using `rumdl` more and more in my organisation, and it greatly helps finding and fixing a wide range of issues.

Last week I found a few Markdown files in a repository that contain some non-UTF8 characters. With the current `rumdl` behavior as of 0.2.75, those files are completely skipped and rumdl doesn't produce any findings for them.
So if you parse only the linter findings (e.g. in JSON ouput), you completely miss those reports and are unaware of the presence of files with encoding issues, so you can't fix them.

### Proposed solution

I'd like `rumdl` to report the presence of non-UTF8 characters as a linting issue instead of fully skipping the file.

If there are only a few non-UTF8 characters those can be reported individually, and actual linting can take place.
If there are too many, produce only a single report and skip the rest of the linting.

### Alternatives considered

Does a dedicated rule make sense for this issue? Reading the file technically happens  before rules are evaluated, but having a placeholder rule (maybe `MD00`?) can make sense.

### Feature category

New lint rule

### Example markdown

```markdown
I can't paste non-UTF8 characters in there...
```

### Contribution

- [x] I would be willing to submit a PR to implement this feature

### Additional context

draft PR is in #903. Please let me know what you think about that before I work on it some more.

**Comment by guillp** [comment]
Many thanks for implementing this! Works like a charm! 💯

**Comment by rvben** [comment]
Thanks for the feedback, @guillp! This indeed is in 0.2.76. Will close this issue.
